### PR TITLE
Fix/char limit translatable

### PR DESCRIPTION
**Pull request by zamandevs**
Besides restriction by minimum enhanced. please @FaraziF test thoroughly.

### CODE REVIEW

**Comment by FaraziF** [comment]
At a time we can select only one restricted type
image

File: assets/js/frontend-form.js [hunk 1]
```diff
@@ -905,20 +908,26 @@
...
+                        numWords = WP_User_Frontend.editorLimit.tinymce.getStats(ed).words;
                     }
 
-                    limit ? $('.mce-path-item.mce-last', ed.container).html( limit_label + numWords +'/'+limit):'';
-
-                    if ( limit && numWords > limit ) {
+                    if ( limit && numWords > limit && 'max' === limit_to ) {
                         WP_User_Frontend.editorLimit.blockTyping(event);
-                        jQuery('.mce-path-item.mce-last', ed.container).html( WP_User_Frontend.content_limit_message( limit_type ) );
+                        WP_User_Frontend.markError( field,'limit');
```
Comment (tareq1988) [inline assets/js/frontend-form.js:923]: fix spacing

File: assets/js/frontend-form.js [hunk 2]
```diff
@@ -905,20 +908,26 @@
...
+                        jQuery('.mce-path-item.mce-last', ed.container).html( wpuf_frontend['word_'+limit_to ] + numWords +'/'+ limit );
+                    }else if ( limit && numWords < limit && 'min' === limit_to ) {
+                        WP_User_Frontend.markError( field,'limit');
```
Comment (tareq1988) [inline assets/js/frontend-form.js:926]: fix spacing

File: assets/js/frontend-form.js [hunk 3]
```diff
@@ -933,18 +942,21 @@
...
                         content_length = self.val().split(' ').length;
                     }
 
-                if ( limit && content_length > limit ) {
-                    self.closest('.wpuf-fields').find('span.wpuf-wordlimit-message').html( WP_User_Frontend.content_limit_message( limit_type ) );
+                if ( limit && content_length > limit && 'max' === limit_to ) {
+                    WP_User_Frontend.content_limit_message( self, limit_type, limit_to, limit )
                     WP_User_Frontend.editorLimit.blockTyping(event);
-                } else {
+                }else if ( limit && content_length < limit && 'min' === limit_to ) {
```
Comment (tareq1988) [inline assets/js/frontend-form.js:956]: You've problems with space in the `else` parenthesis I guess, I saw it multiple times. Everywhere I see the spacing problem. Fix this one and the line below, and make sure it doesn't happen in the future.

File: includes/fields/class-abstract-fields.php [hunk 4]
```diff
@@ -29,6 +29,49 @@ abstract class WPUF_Field_Contract {
     // protected $icon = '';
     protected $icon = 'header';
 
+    /**
+     * @return array[]
+     */
+    public static function common_field() {         return [
```
Comment (tareq1988) [inline includes/fields/class-abstract-fields.php:35]: Fix this one and the indentation of the array.

File: includes/fields/class-field-textarea.php [hunk 5]
```diff
@@ -80,42 +80,46 @@ class="textareafield <?php echo esc_attr( ' wpuf_' . $field_settings['name'] . '
...
-        } ?>
-
-        <?php $this->after_field_print_label();
+                        <?php
+                    }
+                    ?>
+
+                 <?php
+                    $this->help_text( $field_settings );
+                    error_log( print_r( $field_settings, true ) );
```
Comment (tareq1988) [inline includes/fields/class-field-textarea.php:89]: remove error logging

**Review by tareq1988 — CHANGES_REQUESTED** [review]

**Pull request merged**